internal/commands/exec/exec.go

```diff
@@ -205,7 +214,9 @@ func login(config ExecConfig) (*api.Client, string, error) {
 
 	// Now update the client so that it will read the ACL token we just fetched.
 	config.ConsulConfig.Token = token
-	newClient, err := api.NewClient(&config.ConsulConfig)
+	consulConfig := &config.ConsulConfig
+	consulConfig.Namespace = config.GatewayConfig.Namespace
```

Comment:
Same as above.

Reply:
Yeah, good call, I can just change this to `config.ConsulConfig.Namespace = config.GatewayConfig.Namespace` and keep the old `NewClient` code as it was previously.